packages/ui/src/__tests__/molecules/Search/Search.test.tsx

```diff
@@ -59,4 +71,34 @@ describe('Search component', () => {
     expect(input.value).toBe('')
     expect(closeIcon).not.toBeInTheDocument()
   })
+
+  it('calls handleSearchValue with debounced value', () => {
+    const debounceDelay = 300
+
+    render(
+      <Search
+        {...mockSearch}
+        debounceDelay={debounceDelay}
+        handleSearchValue={mockHandleSearchValue}
+      />
+    )
+
+    const input: HTMLInputElement = screen.getByTestId('input-search-test')
+    fireEvent.change(input, { target: { value: 'debounce value test' } })
+    act(() => {
+      vi.advanceTimersByTime(debounceDelay)
+    })
+    expect(mockHandleSearchValue).toHaveBeenCalledWith('debounce value test')
+  })
+
+  it('uses default debounce delay when value not provided', () => {
+    render(<Search {...mockSearch} handleSearchValue={mockHandleSearchValue} />)
```

Comment:
The value is provided by {...mockSearch} in this case, maybe you can delete the debounceDelay in the mockSearch if there is a default value.